paddle/cinn/operator_fusion/backend/pattern_fuser.cc

```diff
@@ -141,71 +131,263 @@ std::vector<FusionOp> ReduceTreeTrivialTransformRecursive(
 }
 
 /// End: Tmp Transform Operation for reduce tree
+///
+struct FusionOp2Expr {
+  std::vector<ir::Expr> operator()(const TrivialOp& op) {
+    return {op.GetFuncBody()};
+  }
+  std::vector<ir::Expr> operator()(const ReduceOp& op) {
+    const auto& t_r = SplitReduceOp(op);
+    return {t_r.first.GetFuncBody(), t_r.second.GetFuncBody()};
+  }
+};
 
-std::vector<FusionOp> GetFusionOpFromPattern(
+std::vector<ir::Expr> GetExprFromPattern(
     const StmtPattern<BackendStage>& pattern);
 
-struct FusionOpGetter {
-  std::vector<FusionOp> operator()(
+static std::vector<ir::Var> GetAllForIters(const ir::Expr& expr) {
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      ChildFors;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      ChildScheduleBlockRealizes;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      FindFather;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      IsFor;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      ScheduleBlockRealizeIsNotInit;
+  const auto& all_father_fors =
+      (ChildScheduleBlockRealizes * ScheduleBlockRealizeIsNotInit *
+       FindFather(expr) * IsFor)(expr);
+  std::vector<ir::Var> vars;
+  for (const auto& for_expr : all_father_fors) {
+    vars.push_back(for_expr.As<ir::For>()->loop_var);
+  }
+  VLOG(4) << "GetAllForIters : " << expr
+          << "\n var is : " << utils::Join(vars, ",");
+  return vars;
+}
+
+ir::Expr UnSqueezeExpr(const ir::Expr& expr,
+                       const std::vector<int> padding_vec) {
+  using cinn::hlir::framework::pir::trivial_fusion_detail::AppendBound;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      ChildFors;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      ChildRootScheduleBlockRealizes;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      ChildScheduleBlockRealizes;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::ExprSetFinderUtils::
+      IsForIterVar;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::
+      ExprTransformerUtils::ReplaceVarTransformer;
+  using cinn::hlir::framework::pir::trivial_fusion_detail::
+      ExprTransformerUtils::UnsqueezeForTransformer;
+  VLOG(4) << "UnSqueezeExpr: " << expr
+          << "\npadding vector: " << utils::Join(padding_vec, ", ");
+  const auto& vars_in_expr = AppendBound(GetAllForIters(expr), expr);
+  // get the all vars.
+  int counter = 0;
+  auto gen_next_name = [&counter]() {
```

Comment:
```suggestion
  auto GenNextName = [&counter]() {
```
小写+下换线一般是变量名，函数名应该要大写？另外counter是局部计数，在全局存在同名场景，这个符合实现预期么？

Reply:
下一个PR修改～